opensearch/src/main/java/org/opensearch/sql/opensearch/mapping/IndexMapping.java

```diff
@@ -50,55 +59,42 @@ public int size() {
    * @return field type in string. Or null if not exist.
    */
   public String getFieldType(String fieldName) {
-    return fieldMappings.get(fieldName);
-  }
-
-  /**
-   * Get all field types and transform raw string type to expected type.
-   *
-   * @param transform transform function to transform field type in string to another type
-   * @param <T> expected field type class
-   * @return mapping from field name to field type
-   */
-  public <T> Map<String, T> getAllFieldTypes(Function<String, T> transform) {
-    return fieldMappings.entrySet().stream()
-        .collect(Collectors.toMap(Map.Entry::getKey, e -> transform.apply(e.getValue())));
+    if (!fieldMappings.containsKey(fieldName)) {
+      return null;
+    }
+    return fieldMappings.get(fieldName).typeName();
   }
 
   @SuppressWarnings("unchecked")
-  private Map<String, String> flatMappings(Map<String, Object> indexMapping) {
-    ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
-
-    flatMappings(
-        ((Map<String, Object>) indexMapping.getOrDefault("properties", emptyMap())),
-        "",
-        builder::put);
-    return builder.build();
+  private Map<String, OpenSearchDataType> parseMapping(Map<String, Object> indexMapping) {
+    Map<String, OpenSearchDataType> result = new HashMap<>();
+    Map<String, Object> mappingInfo = null;
+    if (indexMapping.containsKey("properties")) {
+      mappingInfo = (Map<String, Object>)indexMapping.get("properties");
+    } else if (indexMapping.containsKey("fields")) {
+      mappingInfo = (Map<String, Object>)indexMapping.get("fields");
+    }
+    if (mappingInfo != null) {
+      mappingInfo.forEach((k, v) -> {
+        var innerMap = (Map<String, Object>)v;
+        // TODO: confirm that only `object` mappings can omit `type` field.
+        var type = ((String) innerMap.getOrDefault("type", "object")).replace("_", "");
```

Comment:
question, why replace?

Reply:
Good question
To parse `OpenSearchDataType.Type` - enum values are in CamelCase without underscores